pkg/operator/bootstrap.go

```diff
@@ -337,3 +334,20 @@ func getPlatformManifests(manifests []manifest, platformName string, lbType conf
 	}
 	return platformManifests
 }
+
+// getCloudPlatformManifests returns a the manifests to be generated for cloud platforms.
+// Cloud default Load Balancers are still in use so only the coredns manifests need to
+// be generated.
+func getCloudPlatformManifests(manifests []manifest, platformName string) []manifest {
+	platformManifests := append(manifests,
+		manifest{
+			name:     "manifests/cloud-platform-alt-dns/coredns.yaml",
+			filename: platformName + "/manifests/coredns.yaml",
+		},
+		manifest{
+			name:     "manifests/cloud-platform-alt-dns/coredns-corefile.tmpl",
+			filename: platformName + "/static-pod-resources/coredns/Corefile.tmpl",
+		},
+	)
+	return platformManifests
+}
```

Comment:
This function looks like copy-pasta of `func getPlatformManifests` above with one if-statement removed. Could we instead use one function for everything and make it accept proper set of parameters? Right now I have a feeling that it's some kind of a "very much in rush" work that copies code around just so that it works, but in all this we forget that this code has to be maintainable in the future by probably other people that author it today

Reply:
@mkowalski yes, I do want to complete this feature in the most efficient way possible. I am not in a rush to such an extent that I provide code that is hard to maintain.
I definitely did consider the point your brought up, but concluded differently. I assumed it would be easier if there is a separate method dealing with just the cloud platform manifests. I am happy to implement a common method.

Reply:
To explain my reasoning so that it's not just a pointless complain... I feel like having separate `getPlatformManifests` and `getCloudPlatformManifests` is going to hurt us in a long run because we have platforms that are on-prem and cloud at the same time depending how you use them and where - best example can be vSphere which uses k8s cloud provider but at the same time it looks much more like bare-metal than e.g. GCP

So only for that reason, that "cloud" is not always the same, I'd rather see e.g. `configv1.PlatformLoadBalancerType` getting a value `"none"` and then inside `func getPlatformManifests` you only need a param to have `manifests/on-prem` or `manifests/cloud-platform-alt-dns`

Probably language would help us if we could have a function accept arbitrary number of arbitrarily-typed parameters but we don't have that. But very quickly crafted (take it with a grain of salt) function I see would be

```
func getPlatformManifests(manifests []manifest, platformName string, manifestLocation string, lbType configv1.PlatformLoadBalancerType) []manifest {
	platformManifests := append(manifests,
		manifest{
			name:     "manifests/" + manifestLocation + "/coredns.yaml",
			filename: platformName + "/manifests/coredns.yaml",
		},
		manifest{
			name:     "manifests/" + manifestLocation + "/coredns-corefile.tmpl",
			filename: platformName + "/static-pod-resources/coredns/Corefile.tmpl",
		},
	)
	if lbType == configv1.LoadBalancerTypeOpenShiftManagedDefault || lbType == "" {
		[...]
	}
	return platformManifests
}
```

and then from GCP you call it

```
manifests = getPlatformManifests(manifests, strings.ToLower(string(configv1.GCPPlatformType)), "cloud-platform-alt-dns", "none")
```

and from real on-prem you call it

```
manifests = getPlatformManifests(manifests, strings.ToLower(string(configv1.BareMetalPlatformType)), "on-prem", lbType)
```

I already dislike that inside `if infra.Status.PlatformStatus.VSphere != nil {` we do a bunch of manual `append(manifests,...` what replicates what we have in `getPlatformManifests`. But that's already there and I can only make a note to myself to maybe clean it up a bit the next time I touch vSphere...

Anyway, I just hope we can avoid the next platform to do additional things by copy-pasting and end up with a bunch of `getPlatformXYZ[...]Manifests` functions

Reply:
I hadn't considered the VSphere case. I had update this fix before your comment in a way that didn't change the signature of getPlatformManifests(). I hope this version also works.

Reply:
> Anyway, I just hope we can avoid the next platform to do additional things by copy-pasting and end up with a bunch of getPlatformXYZ[...]Manifests functions

All on-prem platforms would use the existing method and other cloud platforms like GCP (now), (AWS and Azure in 4.16) would have all used the same new method. Not sure how this conclusion about 1 method per platform was drawn.

Reply:
I am not saying that you would create 1 method per platform, but I can already see things behaving differently between GCP and Azure and yet-another-cloud and it's just a question of time when someone does it.

Already see it with vSphere where instead of simply using the function we have, it does everything in-line ¯\_(ツ)_/¯

Thanks a lot for refactor, now it does look amazingly better!

Reply:
@mkowalski I appreciate the feedback and the review.